cmake/functions/four_c_auto_define_tests.cmake

```
@@ -0,0 +1,230 @@
+# This file is part of 4C multiphysics licensed under the
+# GNU Lesser General Public License v3.0 or later.
+#
+# See the LICENSE.md file in the top-level for license information.
+#
+# SPDX-License-Identifier: LGPL-3.0-or-later
+
+function(_set_up_unit_test_target _target)
+  set(options "")
+  set(oneValueArgs NP THREADS)
+  set(multiValueArgs "")
+  cmake_parse_arguments(
+    _parsed
+    "${options}"
+    "${oneValueArgs}"
+    "${multiValueArgs}"
+    ${ARGN}
+    )
+
+  if(DEFINED _parsed_UNPARSED_ARGUMENTS)
+    message(SEND_ERROR "There are unparsed arguments: ${_parsed_UNPARSED_ARGUMENTS}")
```

```suggestion
    message(FATAL_ERROR "There are unparsed arguments: ${_parsed_UNPARSED_ARGUMENTS}")
```

Should we use a fatal error here? https://cmake.org/cmake/help/v3.16/command/message.html#:~:text=FATAL_ERROR,continue%20processing%2C%20but%20skip%20generation.

We used a fatal error in other places and this would stop the processing and generation